### ISSUE TITLE
which one is used to early stop? eval_metrix or loss

### ISSUE THREAD

**Issue by nickhuangxinyu** [catboost/catboost#1853]
my catboost Regressor is:

`CatBoostRegressor(iterations=2000, depth=5, learning_rate= 0.1, loss_function='RMSE', random_state=100, eval_metric=MyMetric())`

i used inner loss RMSE and custom defined metric MyMetric

when i train my model, i found the output is based on my metric. no loss info printing.

So, if the early stop happening, which one is used to stop? metric or loss?

And, How to printing metric and loss at the same time?

**Comment by kizill** [comment]
Hello!
We use `eval_metric` both for early stopping and trimming model to best point (when `use_best_model=True`)
For more look at official documentation: https://catboost.ai/docs/features/loss-functions-desc.html
